analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializedToDefault.cs

```diff
@@ -139,9 +139,8 @@ private static bool IsValueTypeDefaultValueInitializer(EqualsValueClauseSyntax i
                         {
                             return true;
                         }
-                        else if (initializer.Value is ObjectCreationExpressionSyntax || ImplicitObjectCreationExpressionSyntaxWrapper.IsInstance(initializer.Value))
+                        else if (ObjectCreationFactory.TryCreate(initializer.Value) is { } objectCreation)
```

Comment:
```suggestion
                        else if (ObjectCreationFactory.TryCreate(initializer.Value) is { ArgumentList.Arguments: { Count: > 0 } arguments })
```

Reply:
I'm afraid we can't do that otherwise we will miss the case when ArgumentList is null or the count of the arguments is 0.

Reply:
Good catch. Ignore my comment.